src/Helper/CommandExecutor.php

```diff
@@ -140,21 +140,41 @@ private function readAlternating(array $descriptors)
             stream_set_blocking($descriptor, false);
             $outputs[$key] = '';
         }
+        $nretries = 6;
+        $timeout = 15;
+        //$tmpfile = '/tmp/READALTERNATING_'.date('Y-m-d-H-i-s');
+        //file_put_contents($tmpfile, "Starting ...\n");
         do {
-            $read = $descriptors;
-            $write = null;
-            $except = null;
-            stream_select($read, $write, $except, null);
+            for ($i = 0; $i < $nretries; ++$i) {
+                $read = $descriptors;
```

Comment:
You should define starting values for `$read` and `$nresources` before `for` loop. Like:
```php
...
$nresources = 0;
$read = [];

for ($i = 0; $i < $nretries; ++$i) {
...
```

Reply:
Upon inspection, I've got the point of initializing `$read` outside the loop.